reactioncommerce/reaction
# [WIP] Fix red dots appearing on products in marketplace

Resolves #2868 

The variant validation for the `inventoryQuantity` field was causing this issue, I noted that we were using the `ReactionProduct.checkChildVariants` in the custom validation which was causing the validation to change from false to true depending on whether the variant has child variants instead of `ReactionProduct.checkInventoryVariants` which checks if the quantity field for the inventory has been filled. 

**Note:** Product variants that have the inventory quantity field empty will fail the variant validation in both the second shop and marketplace, resulting to the red dot.

**How To Test**
1. Login as as Admin
2. Create a second shop
3. Login as the second shop owner
4. Create a product
5. Observe that any product that is failing validation in the second shop is also failing the validation in the marketplace.
